### PR TITLE
fix: ordinals recover tx

### PR DESCRIPTION
**Pull request by abdulhaseeb4239**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Enhancement
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background
The recover ordinal flow was not working properly after the migration of ordinal transaction logic. 

Issue Link: https://linear.app/xverseapp/issue/ENG-2690/fix-ordinal-recover-logic
Context Link (if applicable):

# 🔄 Changes
Does this PR introduce a breaking change?

- [ ] Yes, Incompatible API changes
- [ ] No, Adds functionality (backwards compatible)
- [x] No, Bug fixes (backwards compatible)

Changes:
- added a new function that fetches ordinals utxos in BTC address
- pass ordinals utxos to sign ordinals transaction function

Impact:
- Send Ordinal, Recover Ordinal
- We need to test this PR with following PRs on mobile and extension
https://github.com/secretkeylabs/xverse/pull/1165
https://github.com/secretkeylabs/xverse-web-extension/pull/552

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: transactions/btc.ts [hunk 1]
```diff
@@ -501,12 +508,13 @@ export async function getBtcFeesForOrdinalTransaction(feeParams: {
   const address = isRecover ? btcAddress : ordinalsAddress;
   const addressUtxos = await btcClient.getUnspentUtxos(address);
   const ordUtxo = getOrdinalUtxo(addressUtxos, ordinal);
+  const ordinalsUtxos = await getOrdinalsUtxos(btcAddress);
```
Comment (teebszet) [inline transactions/btc.ts:511]:
```suggestion
  const addressOrdinalsUtxos = await getOrdinalsUtxos(btcAddress);
```

File: transactions/btc.ts [hunk 2]
```diff
@@ -501,12 +508,13 @@ export async function getBtcFeesForOrdinalTransaction(feeParams: {
...
   if (!ordUtxo) {
     // TODO: Throw error and not just the code
     // eslint-disable-next-line @typescript-eslint/no-throw-literal
     throw new ResponseError(ErrorCodes.OrdinalUtxoNotfound).statusCode;
   }
-  return getBtcFeesForOrdinalSend(recipientAddress, ordUtxo, btcAddress, network, addressUtxos, feeMode, feeRateInput);
+  return getBtcFeesForOrdinalSend(recipientAddress, ordUtxo, btcAddress, network, ordinalsUtxos, feeMode, feeRateInput);
```
Comment (teebszet) [inline transactions/btc.ts:517]:
```suggestion
  return getBtcFeesForOrdinalSend(recipientAddress, ordUtxo, btcAddress, network, addressOrdinalsUtxos, feeMode, feeRateInput);
```

File: transactions/btc.ts [hunk 3]
```diff
@@ -901,6 +909,7 @@ export async function signOrdinalTransaction(ordinalTxParams: {
   const address = isRecover ? btcAddress : ordinalsAddress;
   const addressUtxos = await btcClient.getUnspentUtxos(address);
   const ordUtxo = getOrdinalUtxo(addressUtxos, ordinal);
+  const ordinalsUtxos = await getOrdinalsUtxos(btcAddress);
```
Comment (teebszet) [inline transactions/btc.ts:912]:
```suggestion
  const addressOrdinalsUtxos = await getOrdinalsUtxos(btcAddress);
```

File: transactions/btc.ts [hunk 4]
```diff
@@ -913,7 +922,7 @@ export async function signOrdinalTransaction(ordinalTxParams: {
     accountIndex,
     seedPhrase,
     network,
-    addressUtxos,
+    ordinalsUtxos,
```
Comment (teebszet) [inline transactions/btc.ts:925]:
```suggestion
    addressOrdinalsUtxos,
```

**Review by teebszet — APPROVED** [review]

**Comment by teebszet** [comment]
@abdulhaseeb4239 I already merged this one, but for some reason the tests are failing now.

I checked out this branch and ran npm test, which spends some time on 

     × can calculate fee for ordinal send transaction 6399ms
     × can sign ordinal send transaction 5222ms
     
then fails.

are these tests calling remote APIs that you need to mock out? why did the tests pass before, but fail now, and why does it spend so long on them?

**Comment by abdulhaseeb4239** [comment]
> 

it is taking long because we are making four API calls in one function. I think it is failing because UTXO set is different. Maybe we need to mock a fixed set of UTXOs.

**Pull request merged**